controller/src/main/java/io/pravega/controller/store/stream/StreamMetadataStore.java

```diff
@@ -1098,4 +1098,9 @@
      * @return CompletableFuture which indicates completion of processing.
      */
     CompletableFuture<Void> deleteWaitingRequestConditionally(String scope, String stream, String processorName, OperationContext context, ScheduledExecutorService executor);
+
+    /**
+     * This method initializes any piece of metadata that is necessary for the correct operation of the Controller.
+     */
+    void initializeMetadataStore();
```

Comment:
Out of curiosity: say that we want to destroy a cluster, would we need code to also delete the initialization metadata? Maybe that would be done externally, via a script or something.

Reply:
Changed the approach according to the suggestions from @shiveshr. Now the buckets (including `ownership` znode) get initialized in `StreamCutService`.